File: my-submissions/4135-concatenate-non-zero-digits-and-multiply-by-sum-i/solution.java
class Solution {
    public long sumAndMultiply(int n) {
        if (n == 0) {
            return 0;
        }

        List<Integer> arr = new ArrayList<>();

        while (n > 0) {
            int d = n % 10;
            if (d > 0) {
                arr.add(d);
            }
            n /= 10;
        }

        long sum = 0;
        for (int d : arr) {
            sum += d;
        }

        long value = 0;
        long tens = 1;

        for (int d : arr) {
            value += (long)d * tens;
            tens *= 10;
        }

        return value * sum;
    }
}

File: my-submissions/4135-concatenate-non-zero-digits-and-multiply-by-sum-i/solution.py
class Solution:
    def sumAndMultiply(self, n: int) -> int:
        if n == 0:
            return 0
        arr = []
        while n:
            d = n % 10
            if d > 0:
                arr.append(d)
            n = n // 10
        s = sum(arr)
        res = 0
        for i, d in enumerate(arr):
            res += (d * 10 ** i)
        return res * s
